tests/core/scenario/test_scenario_manager.py

```diff
@@ -813,6 +813,50 @@ def test_get_set_primary_scenario():
     assert _ScenarioManager._get_primary(cycle_1) == scenario_2
 
 
+def test_get_primary_scenarios_sorted():
+    scenario_1_cfg = Config.configure_scenario(id="scenario_1", frequency=Frequency.DAILY)
+    scenario_2_cfg = Config.configure_scenario(id="scenario_2", frequency=Frequency.DAILY)
+
```

Comment:
Maybe we can create a scenario that won't be primary to test that the method works fine filtering out these not primary scenarios.
```suggestion

not_primary_scenario = _ScenarioManager._create(scenario_1_cfg, name="not_primary_scenario")
```

Reply:
I had not thought of that case, good one!